crates/ere-miden/src/compile.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{
+    MIDEN_TARGET, MidenProgram,
+    error::{CompileError, MidenError},
+};
+use miden_assembly::Assembler;
+use miden_core::utils::Serializable;
+use miden_stdlib::StdLibrary;
+use std::{fs, path::Path};
+use zkvm_interface::Compiler;
+
+impl Compiler for MIDEN_TARGET {
+    type Error = MidenError;
+    type Program = MidenProgram;
+
+    fn compile(&self, guest_directory: &Path) -> Result<Self::Program, Self::Error> {
+        let dir_name = guest_directory
+            .file_name()
+            .and_then(|name| name.to_str())
+            .ok_or(CompileError::InvalidProgramPath)?;
+
+        let entrypoint = format!("{}.masm", dir_name);
```

Comment:
To make clippy happy
```suggestion
        let entrypoint = format!("{dir_name}.masm");
```